src/olympia/addons/serializers.py

```diff
@@ -211,6 +212,48 @@ class Meta:
                   'release_notes', 'reviewed', 'url', 'version')
 
 
+class CurrentVersionSerializer(SimpleVersionSerializer):
+    def to_representation(self, obj):
+        # If the add-on is a langpack, and `appversion` is passed, try to
+        # determine the latest public compatible version and replace the obj
+        # with the result. Because of the perf impact, only done for langpacks
+        # in the detail API.
```

Comment:
We're not doing anything to lesson the performance impact though, are we?  I don't see any extra caching, etc.  Is anything planned in a follow-up, etc?  Or are we strictly just limiting it by it only being langpacks?

Reply:
Stricly limiting it by it only being langpacks